cmflib/cmfquery.py

```diff
@@ -617,6 +617,46 @@ def get_one_hop_parent_executions(self, execution_id: t.List[int], pipeline_id:
                 list_exec.append(self.store.get_executions_by_id(exec))
         return list_exec
 
+    def get_one_hop_parent_executions_ids(self, execution_id: t.List[int], pipeline_id: str = None) -> t.List[int]:
+        """Get parent execution ids for given execution id
+        Args: 
+           execution_id : Execution id for which parent execution are required
+                          It is passed in list, for example execution_id: [1]  
+           pipeline_id : Pipeline id
+        Return:
+           Returns parent executions for given id
+        """
+        artifacts: t.Optional = self._get_input_artifacts(execution_id)
+        if not artifacts:
+            return None
+
+        exe_ids = []
+
+        for id in artifacts:
+            ids = self._get_executions_by_output_artifact_id(id, pipeline_id)
+            exe_ids.extend(ids)
+        return exe_ids
+
+    def get_executions_with_execution_ids(self, exe_ids: t.List[int]):
+        """For list of execution ids it returns df with "id,Execution_type_name, Execution_uuid"
+        Args:
+            execution ids: List of execution ids.
+        Return:
+            ["id","Execution_type_name","Execution_uuid"]
+        """
+        df = pd.DataFrame()
+        executions = self.store.get_executions_by_id(exe_ids)
+        count = 0
```

Comment:
You can use enumerate instead of using the counter

Reply:
Updated with enumerate